### ISSUE TITLE
Collection.store updates records with an existing id value instead of throwing error

### ISSUE THREAD

**Issue by jordandenison** [rethinkdb/horizon#555]
When running the following code twice:

```
messages.store({
  id: 1,
  from: "bob",
  text: "Hello from RethinkDB"
})
.subscribe(data => console.log('data1 ', data), err => console.log('err1 ', err))
```

the error handler never fires, but the writeFunction/success handler returns:

```
{id: 1, $hz_v$: 0}
{id: 1, $hz_v$: 1}
{id: 1, $hz_v$: 2}
```

where `$hz_v$` is being incremented on each store.  Not sure if this is intentional or not, but the documentation does state "The objects must have unique id values, and must have id values that do not already exist in the collection or an error will be raised."

(Client/Server version 1.0.3)

**Comment by deontologician** [comment]
Yeah, store is supposed to fail, this is acting more like upsert.
